module/zfs/dmu_send.c

```diff
@@ -959,9 +948,11 @@ do_dump(dmu_send_cookie_t *dscp, struct send_range *range)
 			    bp, arc_getbuf_func, &abuf, ZIO_PRIORITY_ASYNC_READ,
 			    zioflags, &aflags, &zb) != 0)
 				return (SET_ERROR(EIO));
-
-			err = dump_spill(dscp, bp, zb.zb_object, abuf->b_data);
-			arc_buf_destroy(abuf, &abuf);
+			if (!dscp->dsc_dso->dso_dryrun) {
+				err = dump_spill(dscp, bp, zb.zb_object,
+				    abuf->b_data);
+				arc_buf_destroy(abuf, &abuf);
+			}
```

Comment:
It looks like this is a bug fix, which we might want to backport to 0.8.  Could you elaborate on how we hit this, and what the impact is?

Reply:
I'd suggest opening a PR with just this fix to make it easy to backport.

Reply:
It was an issue I hit when doing the initial porting work. I'll just revert it and make a mental note.

Reply:
If you look at the code it seems self evident that we're dumping a buffer and destroying an arc_buf when no read has happened: 
```
			arc_buf_t *abuf;
			zbookmark_phys_t zb;
			ASSERT3U(range->start_blkid, ==, DMU_SPILL_BLKID);
			zb.zb_objset = dmu_objset_id(dscp->dsc_os);
			zb.zb_object = range->object;
			zb.zb_level = 0;
			zb.zb_blkid = range->start_blkid;

			if (!dscp->dsc_dso->dso_dryrun && arc_read(NULL, spa,
			    bp, arc_getbuf_func, &abuf, ZIO_PRIORITY_ASYNC_READ,
			    zioflags, &aflags, &zb) != 0)
				return (SET_ERROR(EIO));
			if (!dscp->dsc_dso->dso_dryrun) {
				err = dump_spill(dscp, bp, zb.zb_object,
				    abuf->b_data);
				arc_buf_destroy(abuf, &abuf);
			}
```